pkg/asset/machines/aws/machinesets.go

```diff
@@ -34,27 +45,35 @@ func MachineSets(clusterID string, region string, subnets icaws.Subnets, pool *t
 		if int64(idx) < total%numOfAZs {
 			replicas++
 		}
-		subnet, ok := subnets[az]
-		if len(subnets) > 0 && !ok {
-			return nil, errors.Errorf("no subnet for zone %s", az)
-		}
 
-		publicSubnet := subnet.Public
-		instanceType := mpool.InstanceType
 		nodeLabels := make(map[string]string, 3)
 		nodeTaints := []corev1.Taint{}
+		instanceType := mpool.InstanceType
+		publicSubnet := false
+		subnetID := ""
+		if len(in.Subnets) > 0 {
+			subnet, ok := in.Subnets[az]
+			if !ok {
+				return nil, errors.Errorf("no subnet for zone %s", az)
+			}
+			publicSubnet = subnet.Public
+			subnetID = subnet.ID
+		}
 
-		if pool.Name == types.MachinePoolEdgeRoleName {
-			// edge pools typically do not receive the same workloads between
-			// different zoneGroups, thus the installer will discover preferred
-			// instance based on the installer's preferred instance lookup.
-			if subnet.PreferredEdgeInstanceType != "" {
-				instanceType = subnet.PreferredEdgeInstanceType
+		if in.Pool.Name == types.MachinePoolEdgeRoleName {
+			// edge pools not share same instance type and regular cluster workloads.
+			// The instance type is selected based in the offerings for the location.
+			// The labels and taints are set to prevent regular workloads.
+			// https://github.com/openshift/enhancements/blob/master/enhancements/installer/aws-custom-edge-machineset-local-zones.md
+			zone := in.Zones[az]
+			if zone.PreferredInstanceType != "" {
+				instanceType = zone.PreferredInstanceType
 			}
 			nodeLabels = map[string]string{
-				"node-role.kubernetes.io/edge":    "",
-				"machine.openshift.io/zone-type":  subnet.ZoneType,
-				"machine.openshift.io/zone-group": subnet.ZoneGroupName,
+				"node-role.kubernetes.io/edge":          "",
+				"machine.openshift.io/zone-type":        zone.Type,
+				"machine.openshift.io/zone-group":       zone.GroupName,
+				"machine.openshift.io/parent-zone-name": zone.ParentZoneName,
```

Comment:
The label `machine.openshift.io/parent-zone-name` wasn't covered in the enhancement, although the parent zone is used to choose affinity in the NAT Gateway in the private route tables.
For a user use case, they can consume these labels in the deployments to create an affinity in client-servers architectures by deploying "servers" in zones in the Region (example us-east-1f), and clients in the Local Zones (example us-east-1-nyc-1a) - and in the future Wavelength Zones.

It also can help, users/support/admins, to understand why the installer associated the private subnet (example `us-east-1-nyc-1a`) in the "private route table X" (example private route table for zone `us-east-1f`) when deploying in full zones.

@JoelSpeed @patrickdillon objections in exposing this label?

Reply:
I will keep this thread opened until I update the EP with the latest information from this PR.

Reply:
Can you provide a complete example of what these will look like once populated? How will the zone information look when including all of the standard kube zone information and the new ones you're adding here?

Reply:
here are the node labels:
```
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/instance-type=c5d.2xlarge
                    beta.kubernetes.io/os=linux
                    failure-domain.beta.kubernetes.io/region=us-west-2
                    failure-domain.beta.kubernetes.io/zone=us-west-2-den-1a
                    kubernetes.io/arch=amd64
                    kubernetes.io/hostname=ip-10-0-134-122.us-west-2.compute.internal
                    kubernetes.io/os=linux
                    machine.openshift.io/parent-zone-name=us-west-2d
                    machine.openshift.io/zone-group=us-west-2-den-1
                    machine.openshift.io/zone-type=local-zone
                    node-role.kubernetes.io/edge=
                    node-role.kubernetes.io/worker=
                    node.kubernetes.io/instance-type=c5d.2xlarge
                    node.openshift.io/os_id=rhcos
                    topology.ebs.csi.aws.com/zone=us-west-2-den-1a
                    topology.kubernetes.io/region=us-west-2
                    topology.kubernetes.io/zone=us-west-2-den-1a
```

Reply:
Ack ok, I think I understand, I have no object, lets get the EP updated to reflect the change if possible

Reply:
Thanks, Joel! 

> lets get the EP updated

yes, for sure; it is in the queue, I am in the process if there's nothing else here (currently I am investigating the longer-than-expected time for the node became ready as an [e2e preflight check](https://redhat-internal.slack.com/archives/C999USB0D/p1689951273676029)).

Reply:
PR opened: https://github.com/openshift/enhancements/pull/1444 . Resolving this thread.